thoth/prescriptions_refresh/handlers/package_downloads.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# thoth-prescriptions-refresh
+# Copyright(C) 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Retrieve the number of downloads from PyPI for a given package"""
+
+
+from datetime import datetime
+from google.cloud import bigquery
+import logging
+import matplotlib.pyplot as plt
+import os
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Tuple
+
+if TYPE_CHECKING:
+    from thoth.prescriptions_refresh.prescriptions import Prescriptions
+
+_LOGGER = logging.getLogger(__name__)
+
+_PACKAGE_DOWNLOADS_PRESCRIPTION_NAME = "package_downloads.yaml"
+_PACKAGE_DOWNLOADS_PRESCRIPTION_CONTENT = """
+units:
+  wraps:
+  - name: {prescription_name}
+    type: wrap
+    should_include:
+      adviser_pipeline: true
+    match:
+      state:
+        resolved_dependencies:
+        - name: {package_name}
```

Comment:
It looks like this prescription works solely on packages. It might be an interesting observation to see stats also on version level - e.g. package `foo` is very popular in the community, `foo=1.0.0` is very popular and `foo==1.2.0` is not that popular. This might indicate that the community prefers `1.0.0` for some reason (for example stability) or `foo==1.2.0` is too fresh to recommend. I think we could elaborate here a little bit more to provide better recommendations based on download stats (preferring more downloaded versions for the target environment - see https://github.com/thoth-station/prescriptions-refresh-job/issues/42). However, the first implementation could just inform users. 👍🏻

Reply:
I see, thank you for this remark! I will change the prescription accordingly :+1:

Reply:
Do you think it would be better to write a prescription per package and to give more details about each version popularity in the message or to create separate prescriptions for each package version?

Reply:
It might be interesting to have one prescription for a package and then, in addition to that, prescriptions on the version level.

Reply:
Corrected by https://github.com/thoth-station/prescriptions-refresh-job/pull/65/commits/1405941f2beda55b2271060387227398ed03b2b2